frontend/views/containers/chatroom/ChatroomMixin.js

```diff
@@ -61,30 +61,36 @@ const chatroom: Object = {
       'generalChatRoomId',
       'globalProfile',
       'isJoinedChatRoom',
-      'isPrivateChatRoom'
+      'isPrivateChatRoom',
+      'ourContactProfiles',
+      'isDirectMessage',
+      'mailboxContract',
+      'ourUsername',
+      'directMessageIDFromUsername',
+      'usernameFromDirectMessageID'
     ]),
     ...mapState(['currentGroupId']),
     summary (): Object {
       if (!this.isJoinedChatRoom(this.currentChatRoomId)) {
-        const joiningChatRoomId = sbp('okTurtles.data/get', 'JOINING_CHATROOM_ID')
-        return !joiningChatRoomId
-          ? this.ephemeral.loadedSummary || {}
-          : { ...this.ephemeral.loadedSummary, joined: joiningChatRoomId === this.currentChatRoomId }
+        const joined = sbp('chelonia/contract/isSyncing', this.currentChatRoomId)
+        return Object.assign(this.ephemeral.loadedSummary || {}, { joined })
       }
 
-      const { name, type, description, creator, picture, privacyLevel } = this.currentChatRoomState.attributes
+      const { name, type, description, creator, privacyLevel } = this.currentChatRoomState.attributes
+      const partnerUsername = this.usernameFromDirectMessageID(this.currentChatRoomId)
+      const partner = this.ourContactProfiles[partnerUsername]
+      const partnerName = partner ? partner.displayName || partner.username : ''
 
       return {
         type,
-        title: name,
+        title: type === CHATROOM_TYPES.INDIVIDUAL ? partnerName : name,
         description,
-        routerBack: type === CHATROOM_TYPES.INDIVIDUAL ? '/messages' : '/group-chat',
-        private: this.currentChatRoomState.attributes.privacyLevel === CHATROOM_PRIVACY_LEVEL.PRIVATE,
+        private: privacyLevel === CHATROOM_PRIVACY_LEVEL.PRIVATE,
         privacyLevel,
         general: this.generalChatRoomId === this.currentChatRoomId,
         joined: true,
-        creator,
-        picture
+        picture: type === CHATROOM_TYPES.INDIVIDUAL ? partner?.picture : undefined,
```

Comment:
So... for this PR, based on the conversation we had on Slack about merging GMs & DMs, will you be updating this PR to do that? Or will you be leaving that for the other PR?

Because everything with `CHATROOM_TYPES.INDIVIDUAL` would need to be changed, and I want to make sure we don't forget about that

Reply:
Yeah. I am going to implement that in the #1463. Removing `dms` and `gms` in the mailbox contract.

By the way, even though we merge `dms` and `gms`, why do you think `CHATROOM_TYPES.INDIVIDUAL` need to be changed? Also we don't check `type` this way in the #1463, but I think there could be comparison between `type` and `CHATROOM_TYPES.INDIVIDUAL` anyway.

Reply:
> By the way, even though we merge dms and gms, why do you think CHATROOM_TYPES.INDIVIDUAL need to be changed? 

Because if we are merging `dms` and `gms` - all chatroom types are of type `GROUP`. So to simplify you can also remove `CHATROOM_TYPES` entirely.

Reply:
To be honest, I don't think so.
There are two chatroom types; `CHATROOM_TYPES.INDIVIDUAL` and `CHATROOM_TYPES.GROUP`.
Chatroom of type `GROUP` is accessible only when they joined the specific group.
But chatroom of type `INDIVIDUAL` is accessible no matter they joined that group or not, just like DMs and GMs in Discord.

We can filter those chatrooms from group contract and mailbox contract.
group.chatRooms is listing the chatrooms of type `GROUP`, while chatrooms from mailbox are those of type `INDIVIDUAL`.
But I think we should set this difference as an attribute inside the chatroom contract.

Reply:
And also there is one more problem in merging `dms` and `gms` in mailbox contract.
Keys in `dms` are the usernames, while the keys in `gms` are contractID.
```
{
  dms: {
    alexjin: {},
    greg: {},
    sebin: {},
    ...
  },
  gms: {
    21XWnNHxRmhAxBPyr9aPB4VX1THB6d2S8qT1oGLREHNM4mNsuw: {},
    21XWnNVWxapZ9bGxcxqdoqbnSPHdEUccKucVqC5DaKeqKuYXDD: {},
    ...
  }
}
```
User opens DMs by clicking the username, not contractID.
But for the GMs, we can there is no other identifier except contractID.

Do you have any recommended solution for this?

Reply:
As @taoeffect suggetsed, the key could be like this `usernames.sort().join('|')` which is really awesome.
This will be implemented in the PR #1463 so I will resolve this conversation.